cli/global_state.rs

```diff
@@ -0,0 +1,259 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+use crate::compilers::CompiledModule;
+use crate::compilers::JsCompiler;
+use crate::compilers::JsonCompiler;
+use crate::compilers::TsCompiler;
+use crate::deno_dir;
+use crate::deno_error::permission_denied;
+use crate::file_fetcher::SourceFileFetcher;
+use crate::flags;
+use crate::metrics::Metrics;
+use crate::msg;
+use crate::ops::JsonOp;
+use crate::permissions::DenoPermissions;
+use crate::progress::Progress;
+use deno::ErrBox;
+use deno::Loader;
+use deno::ModuleSpecifier;
+use deno::PinnedBuf;
+use futures::Future;
+use serde_json::Value;
+use std;
+use std::env;
+use std::ops::Deref;
+use std::str;
+use std::sync::atomic::Ordering;
+use std::sync::Arc;
+
```

Comment:
I would have expected a static global variable with the global state...

Reply:
Yeah, makes sense

Reply:
Might be hard to properly do this for tests with single global state but I'll give it a try anyway.

Reply:
Yeah.. It's a good point. Even if you don't make it a static global, I don't think we need to modify the op parameters, rather just add something like `fn ThreadSafeState::global_state() -> ThreadSafeGlobalState`

Reply:
I ended up with `global_state` field in `ThreadSafeState`.